packages/core/src/util/renderer.ts

```diff
@@ -80,11 +82,284 @@ import {
+      propertiesCondition[property],
+      'properties'
+    );
+
+    return all(
+      (prop) =>
+        checkPropertyCondition(
+          nextPropertyConditions,
+          prop,
+          data[property] as Record<string, unknown>
+        ),
+      Object.keys(nextPropertyConditions)
+    );
+  }
+
+  return checkDataCondition(propertiesCondition[property], property, data);
+};
+
+const evaluateCondition = (
+  schema: JsonSchema,
+  data: Record<string, unknown>
+): boolean => {
+  if (has(schema, 'allOf')) {
+    return all(
+      (subschema: JsonSchema) => evaluateCondition(subschema, data),
+      get(schema, 'allOf')
+    );
+  }
+
+  if (has(schema, 'anyOf')) {
+    return any(
+      (subschema: JsonSchema) => evaluateCondition(subschema, data),
+      get(schema, 'anyOf')
+    );
+  }
+
+  if (has(schema, 'oneOf')) {
+    const subschemas = get(schema, 'oneOf');
+
+    let satisfied = false;
+
+    for (let i = 0; i < subschemas.length; i++) {
+      if (satisfied) {
+        return false;
+      }
+
+      satisfied = evaluateCondition(subschemas[i], data);
+    }
+
+    return satisfied;
+  }
+
+  let requiredProperties: string[] = [];
+  if (has(schema, 'required')) {
+    requiredProperties = get(schema, 'required');
+  }
+
+  const requiredCondition = all(
+    (property) => data[property],
+    requiredProperties
+  );
+
+  const propertiesCondition = get(schema, 'properties') as Record<
+    string,
+    unknown
+  >;
+
+  const valueCondition = all(
+    (property) => checkPropertyCondition(propertiesCondition, property, data),
+    Object.keys(propertiesCondition)
+  );
+
+  return requiredCondition && valueCondition;
+};
+
+/**
+ * Go through parent's properties untill the segment is found at the exact level it is defined and check if it is required
+ */
+const extractRequired = (
+  schema: JsonSchema,
+  segment: string,
+  prevSegments: string[]
+) => {
+  let i = 0;
+  let currentSchema = schema;
+  while (
+    i < prevSegments.length &&
+    has(currentSchema, 'properties') &&
+    has(get(currentSchema, 'properties'), prevSegments[i])
+  ) {
+    currentSchema = get(get(currentSchema, 'properties'), prevSegments[i]);
+    ++i;
+  }
+
+  if (i < prevSegments.length) {
+    return false;
+  }
+
+  return (
+    has(currentSchema, 'required') &&
+    (get(currentSchema, 'required') as string[]).includes(segment)
+  );
+};
+
+/**
+ * Check if property's required attribute is set based on if-then-else condition
+ *
+ */
+const checkRequiredInIf = (
+  schema: JsonSchema,
+  segment: string,
+  prevSegments: string[],
+  data: Record<string, unknown>
+): boolean => {
+  const propertiesConditionSchema = get(schema, 'if');
+
+  const condition = evaluateCondition(propertiesConditionSchema, data);
+
+  const ifInThen = has(get(schema, 'then'), 'if');
+  const ifInElse = has(get(schema, 'else'), 'if');
+  const allOfInThen = has(get(schema, 'then'), 'allOf');
+  const allOfInElse = has(get(schema, 'else'), 'allOf');
```

Comment:
This seems a bit too hardcoded. I would prefer if the logic could be generalized. For example `anyOf` and `oneOf` are not checked here.